src/WhoopsRunner.php

```diff
@@ -2,24 +2,70 @@
 
 namespace Franzl\Middleware\Whoops;
 
+use Psr\Http\Message\ServerRequestInterface;
+use Whoops\Handler\JsonResponseHandler;
+use Whoops\Handler\PlainTextHandler;
 use Whoops\Handler\PrettyPageHandler;
+use Whoops\Handler\XmlResponseHandler;
 use Whoops\Run;
 use Zend\Diactoros\Response\StringResponse;
 
 class WhoopsRunner
 {
-    public static function handle($error)
-    {
-        $whoops = new Run();
-        $whoops->pushHandler(new PrettyPageHandler());
-        $whoops->register();
 
+    public static function handle($error, ServerRequestInterface $request)
+    {
         $method = Run::EXCEPTION_HANDLER;
-
+        //load Whoops depending on the accept format
+        $whoops = self::getWhoopsInstance($request);
         ob_start();
         $whoops->$method($error);
         $response = ob_get_clean();
 
         return StringResponse::html($response, 500);
     }
+
+    /**
+     * Returns the whoops instance or create one.
+     *
+     * @param ServerRequestInterface $request
+     *
+     * @return Run
+     */
+    private static function getWhoopsInstance(ServerRequestInterface $request)
+    {
+        $whoops = new Run();
+        if (php_sapi_name() === 'cli') {
+            $whoops->pushHandler(new PlainTextHandler());
+            return $whoops;
+        }
```

Comment:
Hey, a little late now, but it just occurred to me: Does this even make sense? After all, since we're talking about HTTP middleware, this should never be executed in a CLI context, right?